pkg/provider/pods.go

```diff
@@ -404,6 +404,19 @@ func (p *Pod) IsRunAsUserID(uid int64) bool {
 	return *p.Pod.Spec.SecurityContext.RunAsUser == uid
 }
 
+func (p *Pod) IsRunAsNonRoot() bool {
+	for i := range p.Pod.Spec.Containers {
+		securityContext := p.Pod.Spec.Containers[i].SecurityContext
+		if securityContext == nil || securityContext.RunAsNonRoot == nil {
+			continue
+		}
+		if *securityContext.RunAsNonRoot {
+			return true
```

Comment:
You shouldn't return true here, as there might be other containers in the pod that need to be checked...

Reply:
good point! I will change it.